File: springboot-backend/src/main/java/com/example/backend/services/AuthService.java
package com.example.backend.services;

import com.example.backend.dtos.requests.ChangePasswordRequest;
import com.example.backend.dtos.requests.ForgetPasswordRequest;
import com.example.backend.dtos.requests.LoginRequest;
import com.example.backend.dtos.requests.SignupRequest;
import org.springframework.http.ResponseEntity;

public interface AuthService {
    ResponseEntity<?> registerUser(SignupRequest signupRequest);
    ResponseEntity<?> authenticateUser(LoginRequest loginRequest);
    ResponseEntity<?> changePassword(ChangePasswordRequest changePasswordRequest);
    ResponseEntity<?> forgetPassword(ForgetPasswordRequest forgetPasswordRequest);
}

File: angular-frontend/src/app/auth.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { LoginRequest, SignupRequest, ForgetPasswordRequest, ChangePasswordRequest } from './auth-requests';

@Injectable({
  providedIn: 'root'
})
export class AuthService {
  private baseUrl = 'http://localhost:8080/api/auth';

  constructor(private http: HttpClient) {}

  login(loginRequest: LoginRequest): Observable<any> {
    return this.http.post(`${this.baseUrl}/login`, loginRequest);
  }

  register(signupRequest: SignupRequest): Observable<any> {
    return this.http.post(`${this.baseUrl}/register`, signupRequest);
  }

  forgetPassword(forgetPasswordRequest: ForgetPasswordRequest): Observable<any> {
    return this.http.post(`${this.baseUrl}/forget-password`, forgetPasswordRequest);
  }

  changePassword(changePasswordRequest: ChangePasswordRequest): Observable<any> {
    return this.http.post(`${this.baseUrl}/change-password`, changePasswordRequest);
  }
}
